userspace/libsinsp/test/sinsp_with_test_input.h

```diff
@@ -43,9 +43,104 @@ class sinsp_with_test_input : public ::testing::Test
 	sinsp m_inspector;
 
 	void open_inspector(sinsp_mode_t mode = SINSP_MODE_TEST);
-	scap_evt* add_event(uint64_t ts, uint64_t tid, ppm_event_code, uint32_t n, ...);
+
+	template <class ... Ts>
+	void _check_event_params(ppm_event_code event_type, uint32_t n, Ts && ... inputs)
```

Comment:
The function checks:
* that each integer-like argument is properly sized
* that `n` is actually the number of arguments passed